### DISCUSSION TITLE
How to use transactions

### DISCUSSION THREAD

**Discussion by KabudoWiseMan** [sqlc-dev/sqlc#1479] [Q&A] [answered]
I'm new to sqlc and I don't really understand the example "Using transactions" in docs. So how can I create a function that uses a transaction as for database/sql:
```go
function FuncTx() {
	// some code
	tx, err := db.BeginTx(ctx, nil)
	if err != nil {
		log.Fatal(err)
	}
	// do something
	tx.Commit()
}
```

**Comment by Jille** [comment, marked as answer]
You can simply pass the `tx` to the New() function of the generated package.

```
function FuncTx() {
	// some code
	tx, err := db.BeginTx(ctx, nil)
	if err != nil {
		log.Fatal(err)
	}
	q := db.New(tx)
	r, err := q.GetRecord(ctx, 1)
	// ...
	tx.Commit()
}
```

**Reply by Jille** [reply]
I've submitted a PR to update the docs: https://github.com/kyleconroy/sqlc/pull/1583
